cpp/dbscan_test.cpp

```diff
@@ -0,0 +1,84 @@
+#include "cpp/dbscan.hpp"
```

Comment:
@ahans this is the test you initially added to the neighbour search branch, should we keep it?

Reply:
It doesn't add a lot of value, I removed it now.